### ISSUE TITLE
Question : can I just use 'setter' in one component and 'listener' in another ?

### ISSUE THREAD

**Issue by x4080** [pmndrs/jotai#118]
Hi, 

I'm using
```
const [, setcounter] = useAtom(atomJCounter);
```
inside component A

and using
```
const [counter,] = useAtom(atomJCounter);
```
inside component B

but whenever I call setcounter, component A always rerender

How to use setcounter without rerendering component A ?

In recoil, it works out of the box

Thanks

**Comment by dai-shi** [comment]
yes. as jotai provides minimal api, it's slightly different from recoil.
we have [write-only atom](https://github.com/pmndrs/jotai#writeonlyatoms) in core,
and [useUpdateAtom](https://github.com/pmndrs/jotai/blob/master/docs/utils.md#useupdateatom) in utils.

Let us know if they work out for you.

**Comment by x4080** [comment]
@dai-shi Thanks for quick response, I'll take a look at those

**Comment by x4080** [comment]
@dai-shi I changed by code for 'setter' to
```
const setcounter = useUpdateAtom(atomJCounter)
```

and

```
setcounter(lastValue => lastValue + 1)
```

And it seems still rerender the component A, anything I miss ?

Thanks, and sorry to bother you so much

**Comment by x4080** [comment]
Ok, got it - I test the rerender using  (rendered: {++useRef(0).current}) now and it works

**Comment by dai-shi** [comment]
Good that it worked out.

A small note to anyone:
- render and commit are separated in react ("render" in general means "render" + "commit" in react)
- it may render more than once and commit just once
- jotai optimizes commits and you may see some extra renders (meaning, extra "render"s in react, but no extra "render"s in general)
- this is because the upstream use-context-selector (v1.2) which behaves like that
- also React Dev Tools doesn't detect optimized commits well
- it will be eventually improved with use-context-selector (v2) once useMutableSource is available. (v1.2 works properly, no worries.)

**Comment by x4080** [comment]
@dai-shi yes, good pointers that you have there
